drivers/i2c/i2c_ite_it8xxx2.c

```diff
@@ -309,6 +325,386 @@ static void i2c_w2r_change_direction(const struct device *dev)
+	unsigned int key = irq_lock();
+
+	if (enable) {
+		if (config->port == 0) {
+			IT8XXX2_SMB_I2CW2RF |= IT8XXX2_SMB_MAIF |
+					       IT8XXX2_SMB_MAIFI;
+		} else if (config->port == 2) {
+			IT8XXX2_SMB_I2CW2RF |= IT8XXX2_SMB_MBCIF |
+					       IT8XXX2_SMB_MCIFI;
+		}
+	} else {
+		if (config->port == 0) {
+			IT8XXX2_SMB_I2CW2RF &= ~(IT8XXX2_SMB_MAIF |
+						 IT8XXX2_SMB_MAIFI);
+		} else if (config->port == 2) {
+			IT8XXX2_SMB_I2CW2RF &= ~(IT8XXX2_SMB_MBCIF |
+						 IT8XXX2_SMB_MCIFI);
+		}
+	}
+
+	irq_unlock(key);
+}
+
+static void i2c_tran_fifo_write_start(const struct device *dev)
+{
+	struct i2c_it8xxx2_data *data = dev->data;
+	const struct i2c_it8xxx2_config *config = dev->config;
+	uint32_t i;
+	uint8_t *base = config->base;
+	volatile uint8_t *reg_mstfctrl = config->reg_mstfctrl;
+
+	/* Clear start flag. */
+	data->msgs->flags &= ~I2C_MSG_START;
+	/* Enable SMB channel in FIFO mode. */
+	*reg_mstfctrl |= IT8XXX2_SMB_FFEN;
+	/* I2C enable. */
+	IT8XXX2_SMB_HOCTL2(base) = IT8XXX2_SMB_SMD_TO_EN |
+				   IT8XXX2_SMB_I2C_EN |
+				   IT8XXX2_SMB_SMHEN;
+	/* Set write byte counts. */
+	IT8XXX2_SMB_D0REG(base) = data->msgs->len;
+	/*
+	 * bit[7:1]: Address of the target.
+	 * bit[0]: Direction of the host transfer.
+	 */
+	IT8XXX2_SMB_TRASLA(base) = (uint8_t)data->addr_16bit << 1;
+	/* The maximum fifo size is 32 bytes. */
+	data->bytecnt = MIN(data->msgs->len, I2C_FIFO_MODE_MAX_SIZE);
+
+	for (i = 0; i < data->bytecnt; i++) {
+		/* Set host block data byte. */
+		IT8XXX2_SMB_HOBDB(base) = *(data->msgs->buf++);
+	}
+	/* Calculate the remaining byte counts. */
+	data->bytecnt = data->msgs->len - data->bytecnt;
+	/*
+	 * bit[6] = 1b: Start.
+	 * bit[4:2] = 111b: Extend command.
+	 * bit[0] = 1b: Host interrupt enable.
+	 */
+	IT8XXX2_SMB_HOCTL(base) = IT8XXX2_SMB_SRT |
+				  IT8XXX2_SMB_SMCD_EXTND |
+				  IT8XXX2_SMB_INTREN;
+}
+
+static void i2c_tran_fifo_write_next_block(const struct device *dev)
+{
+	struct i2c_it8xxx2_data *data = dev->data;
+	const struct i2c_it8xxx2_config *config = dev->config;
+	uint32_t i, _bytecnt;
+	uint8_t *base = config->base;
+	volatile uint8_t *reg_mstfctrl = config->reg_mstfctrl;
+
+	/* The maximum fifo size is 32 bytes. */
+	_bytecnt = MIN(data->bytecnt, I2C_FIFO_MODE_MAX_SIZE);
+
+	for (i = 0; i < _bytecnt; i++) {
+		/* Set host block data byte. */
+		IT8XXX2_SMB_HOBDB(base) = *(data->msgs->buf++);
+	}
+	/* Clear FIFO block done status. */
+	*reg_mstfctrl |= IT8XXX2_SMB_BLKDS;
+	/* Calculate the remaining byte counts. */
+	data->bytecnt -= _bytecnt;
+}
+
+static void i2c_tran_fifo_write_finish(const struct device *dev)
+{
+	const struct i2c_it8xxx2_config *config = dev->config;
+	uint8_t *base = config->base;
+
+	/* Clear byte count register. */
+	IT8XXX2_SMB_D0REG(base) = 0;
+	/* W/C */
+	IT8XXX2_SMB_HOSTA(base) = HOSTA_ALL_WC_BIT;
+	/* Disable the SMBus host interface. */
+	IT8XXX2_SMB_HOCTL2(base) = 0x00;
+}
+
+static bool i2c_tran_fifo_w2r_change_direction(const struct device *dev)
```

Comment:
You need to be consistent with the use of `bool` or `int` as the return type.  Either change the return type to `int`, or change all the return paths in this function to use `true` and `false`.

It's a smaller change to set the return type to `int`.  Please add a comment that this function returns 0 on a failure to indicate that the current transaction is completed.

Reply:
Done.